### PR TITLE
Fix presence / ignored presence tests

### LINKED ISSUE #589: Failing RealtimePresenceTest presence_get
See [this Travis build failure](https://travis-ci.org/github/ably/ably-java/jobs/698619235#L2232), which appears to be generated [here](https://github.com/ably/ably-java/blob/c9e6933dc63e213ea7e8cb51ab326db406372d58/lib/src/test/java/io/ably/lib/test/realtime/RealtimePresenceTest.java#L3235) but the message in Travis is lacking information.

Might be related to #467

┆Issue is synchronized with this [Jira Bug](https://ably.atlassian.net/browse/SDK-445) by [Unito](https://www.unito.io)

### LINKED ISSUE #869: Enable and fix tests in RealtimePresenceTest
There are 33 of 49 tests that are ignored in RealtimePresenceTest class. This is way too many and ignoring tests leaves parts of code untested.
As the first step, we should enable all ignored tests in this class.
If there are failing tests we need to fix them and add new ones if necessary.

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/ECO-4495) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by sacOO7**
- Fixed #869 
- Fixed #589 
- Implements partial fix for https://github.com/ably/ably-java/issues/871

### CODE REVIEW

File: lib/src/main/java/io/ably/lib/realtime/Presence.java [hunk 1]
```diff
@@ -919,8 +913,8 @@ void onAttached(boolean hasPresence) {
     /**
      * Spec: RTP17g
      */
-    synchronized void enterInternalMembers() {
-        for (final PresenceMessage item: internalPresence.values()) {
+    void enterInternalMembers() {
```
Comment (ttypic) [inline lib/src/main/java/io/ably/lib/realtime/Presence.java:916]: Could you explain why we deleted `synchronized` here?

File: lib/src/main/java/io/ably/lib/realtime/Presence.java [hunk 2]
```diff
@@ -903,11 +902,6 @@ private void failQueuedMessages(ErrorInfo reason) {
      ************************************/
 
     void onAttached(boolean hasPresence) {
-        /* Interrupt get() call => by unblocking presence.waitForSync()*/
-        synchronized (presence) {
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply lib/src/main/java/io/ably/lib/realtime/Presence.java:907]: This was actually added as a part of `no-connection-serial` PR ( It didn't exist before ). At that time, `onAttached` was getting called when channel was actually `ATTACHED`. But, in current case, we are calling `onAttached` before attaching the channel explicitly.

File: lib/src/main/java/io/ably/lib/realtime/Presence.java (hunk 1 above)
Comment (sacOO7) [reply lib/src/main/java/io/ably/lib/realtime/Presence.java:916]: This was added as a part of `no-connection-serial`. Actually, there was no need to add `synchronized` for enteringPresenceMembers. Sometimes, this was causing intermittent waits  for threads trying to call `enteringPresenceMembers`. We don't need thread synchronization here. We are doing the same in ably-go -> https://github.com/ably/ably-go/blob/dd8236fc301eb078803e83efee7615f3ba133341/ably/realtime_presence.go#L187

File: lib/src/main/java/io/ably/lib/realtime/Presence.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sacOO7) [reply lib/src/main/java/io/ably/lib/realtime/Presence.java:907]: If `hasPresence` is true, server sends another `SYNC` message containing all presence messages to be synced. At the end of the sync, we call `notifyAll`. 
If `hasPresence` is false, we are already calling `endSync` which in turn calls `notifyAll`. So, in both cases we are calling `notifyAll`. So we don't need to put it here explicitly.

[the remark this answers is not in the document]
Comment (sacOO7) [reply lib/src/main/java/io/ably/lib/realtime/Presence.java:907]: You can see that [Presence endSync](https://github.com/ably/ably-java/blob/0af4d4e465d777462e92845078f0cd966049541b/lib/src/main/java/io/ably/lib/realtime/Presence.java#L1211) method calls `notifyAll` at the end of the sync.

File: lib/src/main/java/io/ably/lib/realtime/Presence.java (hunk 1 above)
Comment (sacOO7) [reply lib/src/main/java/io/ably/lib/realtime/Presence.java:916]: This was added as a part of no-connection-serial ( probably referred to some old code that had synchronized marked as true ). But, we don't actually need it here.

**Review by ttypic — APPROVED** [review]
LGTM!

**Comment by sacOO7** [comment]
Thanks !

**Pull request merged**